### ISSUE TITLE
How to apply filter on a semantic cache check() function 

### ISSUE THREAD

**Issue by pradeepdev-1995** [redis/redis-vl-python#184]
Here is how i am using semantic cache
```
SemanticCacheObject = SemanticCache(
    name=cache_index,                               # underlying search index name
    prefix=cache_prefix,                            # redis key prefix for hash entries
    redis_url=redis_url,                            # redis connection url string
    distance_threshold=distance_threshold,          # semantic cache distance threshold
    ttl=ttl                                         # time to live each key in cache
)
query_embedding = <embedding of my query>
SemanticCacheObject.check(vector=query_embedding, return_fields=["prompt", "response", "metadata"]):
```
Now I want to filter the responses based on a specific column in the index, before applying the vector search.
Then how to apply filter condition on this semantic cache index in check() function.
Eg: I am expecting something like this
```
SemanticCacheObject.check(vector=query_embedding, return_fields=["prompt", "response", "metadata"], 
filter = <column name> )
```
Is this supported now in redis vl?

**Comment by tylerhutcherson** [comment]
It is not yet, but will be available in the upcoming 0.3.0 release. How would you expect to be able to define the filter options during configuration of the cache? And what about during cache checking? Any additional inputs?

**Comment by pradeepdev-1995** [comment]
@justin-cechmanek Thank you. Will check this
